src/StreamFetcher.js

```diff
@@ -70,11 +70,11 @@ module.exports = class StreamFetcher {
         return fetch(url, {
             headers,
         }).catch((e) => {
-            console.error(`failed to communicate with E&E: ${e}`)
+            logger.error(`failed to communicate with E&E: ${e}`)
             throw e
         }).then((response) => {
             if (response.status !== 200) {
-                debug(
+                logger.debug(
                     'fetch failed with status %d for streamId %s, apiKey %s, sessionToken %s : %o',
                     response.status, streamId, apiKey, sessionToken, response.text(),
```

Comment:
not related to this PR but `response.text()` will be a promise won't it

Reply:
Yeah might want to `await`

Reply:
Updated